### DISCUSSION TITLE
Creating record to collection with relations doesn't update the related record

### DISCUSSION THREAD

**Discussion by abbjetmus** [pocketbase/pocketbase#5867] [Q&A] [answered]
Hi!

Don't know if this is by design, but adding a record to the users collection that has a reference to the brfs collection through a property `brf` doesn't automatically make the brfs collection record aware of the user record.

Do I understand it correctly that I need to update the brfs collection record so that its users property get the user-id as well?

```js
                const brf = await pb.collection("brfs").getOne(authStore.brf!.id, {
                    expand: "users, brf_user_invitations"
                });
```

I expected this request to retrieve all users, bit it doesn't since brfs collection records was not aware of the new user.

<img width="413" alt="image" src="https://github.com/user-attachments/assets/0e12cf2c-f478-4fc0-b6d5-888597c9120e">

<img width="220" alt="image" src="https://github.com/user-attachments/assets/594bb5d4-b54d-4aa3-853d-f73c5dba2fd8">

**Comment by ganigeorgiev** [comment]
I don't understand what the screenshots are supposed to show.

If your 2 collections have 2 `relation` fields each pointing to the other collection, there is no automatic linking or anything like that, if that's are your expectation. If that's what is going on in your case I'd advice to try to keep your data normalized and keep the `relation` only in a single place (_it could be even a junction collection_) to avoid getting out of sync.

If that's not what you are asking, please export your collections configuration as json (Admin UI > Settings > Export collections) and try to elaborate more clearly what doesn't work and what is the expected behavior.

**Reply by abbjetmus** [reply]
Yes that is the case like point out. I'm new to PB coming from dotnet and entity framework as ORM that is how you do it.
If it doesn't work like my assumption, how do I add related data. In my case a brfs can have multiple invitations one-to-many. I want to be able to request the brf together with all invitations. For this I have to add the invitations and then make a new request to update the brf as well. I guess I should use the back-relations approach from the docs?

**Reply by ganigeorgiev** [reply, marked as answer]
It doesn't work like you expect because you can technically have as many relation fields you want pointing to a single collection (aka. "users" could have "brfsPrimary" and "brfsSecondary" fields for example and you can't just assume that adding a new brfs record will somehow add it to the fields).

It is still not clear to me what is your collections structure, what is an "invitation", what is "brfs" and what you are trying to do in general, but again avoid having denormalized data structure and try to keep the relations in a single place.

There is nothing really special with how you are defining relations in PocketBase and it is the same as if you were using a regular database (_the only thing that PocketBase adds in addition is the support for storing multiple references in a single column, but you don't have to use that and you can always fallback to a traditional [junction table structure](https://en.wikipedia.org/wiki/Associative_entity)_).

You can retrieve a record with its relations (either direct or indirect/back-relations) via the `expand` query param.

I'm sorry, I don't know how to explain it better then what is already written in the documentation.

I don't think I can help you further, but if the above doesn't help, please try to more clearly elaborate on your use case and what you are trying to do to avoid the needless back-and-forth (I cannot read your mind and I have literally no context about your project; see also [XY problem](https://xyproblem.info/)).

**Reply by abbjetmus** [reply]
Thank you for the clarification and awesome work on PB. Looking forward exploring it further. 
I will try not adding back and forth references. As I mention it's just a one-to-many relationship and coming form dotnet code-first where database schema is generated through code, you do like this:

```csharp
public class Course
{
  public int CourseID { get; set; }
  public string Title { get; set; }
  public int Credits { get; set; }
  public int DepartmentID { get; set; }
  public virtual Department Department { get; set; }
}

public class Department
{
   public Department()
   {
     this.Courses = new HashSet<Course>();
   }  
   public int DepartmentID { get; set; }
   public string Name { get; set; }
   public decimal Budget { get; set; }
   public DateTime StartDate { get; set; }
   public int? Administrator {get ; set; }
   public virtual ICollection<Course> Courses { get; set; }
}
```
And you get navigational properties "Courses" in Department and "Department" in Course. This way you retrieve the references in both ways but you only have the forein-key in Course. That's why I thought I needed to add all the ids to brfs, but you solve it through back-relations.
